python/tests/unit/arcticdb/version_store/test_empty_column_type.py

```diff
@@ -5,40 +5,511 @@
-        lmdb_version_store.append("test_can_append_to_empty_column", df_empty_3)
-        expected_result = pd.concat([df_empty_1, df_empty_2, df_empty_3], ignore_index=True)
-        assert_frame_equal(lmdb_version_store.read("test_can_append_to_empty_column").data, expected_result)
+@pytest.fixture(params=["float" + s for s in ["32", "64"]])
+def float_dtype(request):
+    yield request.param
+
+
+# object is for nullable boolean
+@pytest.fixture(params=["object", "bool"])
+def boolean_dtype(request):
+    if request.param == "object":
+        pytest.skip("Nullable booleans are temporary disabled")
+    yield request.param
+
+
+@pytest.fixture(params=["datetime64[ns]"])
+def date_dtype(request):
+    yield request.param
+
+
+class TestCanAppendToEmptyColumn:
+    @pytest.fixture(autouse=True)
+    def create_empty_column(self, lmdb_version_store_static_and_dynamic):
+        lmdb_version_store_static_and_dynamic.write("sym", pd.DataFrame({"col1": 2 * [None]}))
+        yield
+
+    def test_integer(self, lmdb_version_store_static_and_dynamic, int_dtype):
+        df_non_empty = pd.DataFrame({"col1": np.array([1,2,3], dtype=int_dtype)})
+        lmdb_version_store_static_and_dynamic.append("sym", df_non_empty)
+        expected_result = pd.DataFrame({"col1": np.array([0,0,1,2,3], dtype=int_dtype)})
+        assert_frame_equal(lmdb_version_store_static_and_dynamic.read("sym").data, expected_result)
+        assert_frame_equal(
+            lmdb_version_store_static_and_dynamic.read("sym", row_range=[0,2]).data,
+            pd.DataFrame({"col1": np.array([0,0], dtype=int_dtype)})
+        )
+        assert_frame_equal(
+            lmdb_version_store_static_and_dynamic.read("sym", row_range=[2,5]).data,
+            df_non_empty
+        )
+
+    def test_float(self, lmdb_version_store_static_and_dynamic, float_dtype):
+        df_non_empty = pd.DataFrame({"col1": np.array([1,2,3], dtype=float_dtype)})
+        lmdb_version_store_static_and_dynamic.append("sym", df_non_empty)
+        expected_result = pd.DataFrame({"col1": np.array([float("NaN"),float("NaN"),1,2,3], dtype=float_dtype)})
+        assert_frame_equal(lmdb_version_store_static_and_dynamic.read("sym").data, expected_result)
+        assert_frame_equal(
+            lmdb_version_store_static_and_dynamic.read("sym", row_range=[0,2]).data,
+            pd.DataFrame({"col1": np.array([float("NaN"),float("NaN")], dtype=float_dtype)})
+        )
+        assert_frame_equal(
+            lmdb_version_store_static_and_dynamic.read("sym", row_range=[2,5]).data,
+            df_non_empty
+        )
+
+    def test_bool(self, lmdb_version_store_static_and_dynamic, boolean_dtype):
+        df_non_empty = pd.DataFrame({"col1": np.array([True, False, True], dtype=boolean_dtype)})
+        lmdb_version_store_static_and_dynamic.append("sym", df_non_empty)
+        expected_result = pd.DataFrame({"col1": np.array([None, None, True, False, True], dtype=boolean_dtype)})
```

Comment:
This must be incorrect for non-nullable bools?

Reply:
Pandas interprets `None` as `False` if you set the dtype to be boolean. I'm happy to add a separate test if you think it will be easier to understand.

Reply:
Ah OK, its sufficient to add comments to the tests saying this then